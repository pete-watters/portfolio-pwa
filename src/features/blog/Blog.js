import React, { PureComponent } from 'react';
import ReactHtmlParser from 'react-html-parser';
import axios from 'axios';
import CONFIG from 'constants/';
import Page from '../../components/Page';

class Blog extends PureComponent {
  state = { feed: {} } 
  componentDidMount() {
    this.fetchFeed().then(this.setFeed);
  }
  fetchFeed = () => axios.get(`https://us-central1-portfolio-8db27.cloudfunctions.net/api/medium`);
  setFeed = response => {
    this.setState({feed: response});
  } 
  render(){
    const { data } = this.state.feed;
      return(
      <Page>
          {data && data.items.map((post, index) => (
              <article key={index} onClick={() => window.open(post.link, "_blank")} aria-label="blog-post">
                <h1>{post.title}</h1>
                <summary>{ReactHtmlParser(post["content:encoded"])}</summary>
              </article>
            ))
            }
            <footer>
              <a href={CONFIG.SOCIAL.MEDIUM} target="_blank" rel="noopener noreferrer">
                {data && data.description}
              </a>
            </footer>
        </Page>);
  }
}

export default Blog;
