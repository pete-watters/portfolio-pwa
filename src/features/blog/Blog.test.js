import React from 'react';
import { shallow } from 'enzyme';
import Blog from './Blog';

const component = shallow(<Blog />);

describe('Blog', () => {
  it('should render the component correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
