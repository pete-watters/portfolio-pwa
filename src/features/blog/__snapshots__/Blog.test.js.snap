// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog should render the component correctly 1`] = `
<Memo(Page)>
  <footer>
    <a
      href="https://medium.com/@pete_watters"
      rel="noopener noreferrer"
      target="_blank"
    />
  </footer>
</Memo(Page)>
`;
